daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/SortedLookupList.scala

```diff
@@ -68,6 +69,9 @@ object SortedLookupList extends SortedLookupListInstances {
       .toLeft(new SortedLookupList(entries))
   }
 
+  def fromOrderedIterator[X](entries: Iterator[(String, X)]): Either[String, SortedLookupList[X]] =
+    fromOrderedImmArray(entries.to[ImmArray])
```

Comment:
You may want to flip these around so that you're not constructing an `ImmArray` which is immediately thrown away.

`ImmArray#iterator` is basically free.

Reply:
If you take an attentive look, you can see that the `ImmArray`  built in `SortedLookupList.fromOrderedIterator` function actually ends up in the contructor of the `SortedLookupList` in the body of `SortedLookupList.fromOrderedImmArray` function.

The only construction of `ImmArray` that can be avoid is done my `.map(_._1)` method call in the body of  `SortedLookupList.fromOrderedImmArray`. I will take care of that in another PR.